### PR TITLE
OSDOCS#11879: Create cluster-admin during cluster deployment

### PR DESCRIPTION
**Pull request by laubai**
Version(s): 4.16+

Issue:
https://issues.redhat.com/browse/OSDOCS-11879

Link to docs preview:
- Terraform creation: https://81465--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_hcp/terraform/rosa-hcp-creating-a-cluster-quickly-terraform.html
- ROSA Classic - quick terraform cluster: https://81465--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/terraform/rosa-classic-creating-a-cluster-quickly-terraform.html
- STS ROSA Classic: https://81465--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html
- Interactive CLI reference: https://81465--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-interactive-mode-reference.html
- CLI reference: https://81465--ocpdocs-pr.netlify.app/openshift-rosa/latest/cli_reference/rosa_cli/rosa-manage-objects-cli.html

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/rosa-sts-interactive-cluster-creation-mode-options.adoc [hunk 1]
```diff
@@ -26,7 +26,10 @@ The following table describes the interactive cluster creation mode options:
 |Enable the use of Hosted Control Planes.
 
 |`Create cluster admin user`
-|Create a cluster administrator user when you create your cluster using the htpasswd identity provider. The username must not contain `/`, `:`, or `%`. The password must be at least 14 characters (ASCII-standard) without whitespaces.
+a|Create a local administrator user (`cluster-admin`) for your cluster. This automatically configures a htpasswd identity provider for the `cluster-admin` user.
+
+|`Create custom password for cluster admin`
+a|Create a custom password for the `cluster-admin` user, or use a system-generated password. If you create a custom password, the password must be at least 14 characters (ASCII-standard) and contain no whitespace characters. If you do not create a custom password, the system generates a password and displays it the command line output.
```
Comment (AkashKanni) [inline modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:32]: `the system generates a password and displays it the command line output` 
i suggest to replace above with
`the system generates a password and displays it on the command line output`

Comment (laubai) [reply modules/rosa-sts-interactive-cluster-creation-mode-options.adoc:32]: Thanks for the catch!

File: modules/rosa-classic-cluster-terraform-file-creation.adoc [hunk 2]
```diff
@@ -102,11 +101,25 @@ module "rosa-classic" {
   multi_az               = var.multi_az
   create_account_roles   = true
   create_operator_roles  = true
-
+# Optional: Configure a cluster administrator user <.>
```
Comment (marcolan018) [inline modules/rosa-classic-cluster-terraform-file-creation.adoc:104]: in the main.tf
the 
```
version                = "1.6.2"
```
should change to 
```
version                = "1.6.3"
```

File: modules/rosa-hcp-cluster-terraform-file-creation.adoc [hunk 3]
```diff
@@ -97,11 +97,25 @@ module "rosa-hcp" {
   aws_subnet_ids         = var.create_vpc ? var.private_cluster ? module.vpc[0].private_subnets : concat(module.vpc[0].public_subnets, module.vpc[0].private_subnets) : var.aws_subnet_ids
   create_account_roles   = true
   create_operator_roles  = true
+# Optional: Configure a cluster administrator user <.>
```
Comment (marcolan018) [inline modules/rosa-hcp-cluster-terraform-file-creation.adoc:100]: for hcp module
the version 1.6.2 should change to 1.6.3

Comment (laubai) [reply modules/rosa-hcp-cluster-terraform-file-creation.adoc:100]: Thanks @marcolan018  - made that change on line 88 and also line 40 of this file.

File: modules/rosa-create-objects.adoc [hunk 4]
```diff
@@ -204,6 +204,15 @@ a|--cluster-name <cluster_name>
 |--controlplane-iam-role <arn>
 |The ARN of the IAM role to attach to control plane instances.
 
+|--create-cluster-admin
+|Optional. As part of cluster creation, create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an htpasswd identity provider for the `cluster-admin` user. Optionally, use the `--cluster-admin-user` and `--cluster-admin-password` options to specify the username and password for the administrator user. Omitting these options automatically generates the credentials and displays their values as terminal output.
```
Comment (dfitzmau) [inline modules/rosa-create-objects.adoc:208]:
```suggestion
|Optional parameter. As part of cluster creation, create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an `htpasswd` identity provider for the `cluster-admin` user. Optionally, use the `--cluster-admin-user` and `--cluster-admin-password` options to specify the username and password for the administrator user. Omitting these options automatically generates the credentials and displays their values as terminal output.
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 5]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
 ? Deploy cluster with Hosted Control Plane (optional): No
 ? Create cluster admin user: Yes <2>
-? Username: user-admin <2>
-? Password: [? for help] *************** <2>
-? OpenShift version: 4.17.0 <3>
-? Configure the use of IMDSv2 for ec2 instances optional/required (optional): <4>
-I: Using arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Installer-Role for the Installer role <5>
+? Create custom password for cluster admin: No <3>
+I: cluster admin user is cluster-admin
+I: cluster admin password is abcd1-eFgh2-jKLm3-NopQ4
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:227]: See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-reference

I had the security team contact me in the past about using password syntax that represents real passwords.

Screenshot from 2024-10-02 12-05-26

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 6]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
+? OpenShift version: 4.17.0 <4>
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:228]:
```suggestion
? OpenShift version: <openshift_version> <4>
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 7]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
+   rosa create cluster --cluster-name <cluster_name> --role-arn arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Installer-Role --support-role-arn arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Support-Role --master-iam-role arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-ControlPlane-Role --worker-iam-role arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-Worker-Role --operator-roles-prefix <cluster_name>-<random_string> --region us-east-1 --version 4.17.0 --additional-compute-security-group-ids sg-0e375ff0ec4a6cfa2 --additional-infra-security-group-ids sg-0e375ff0ec4a6cfa2 --additional-control-plane-security-group-ids sg-0e375ff0ec4a6cfa2 --replicas 2 --machine-cidr 10.0.0.0/16 --service-cidr 172.30.0.0/16 --pod-cidr 10.128.0.0/14 --host-prefix 23 <15>
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:258]: Would moving the example to the <15> callout be better?

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 8]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
-<2> When creating your cluster, you can create a local administrator user for your cluster. Selecting `Yes` then prompts you to create a user name and password for the cluster admin. The user name must not contain `/`, `:`, or `%`. The password must be at least 14 characters (ASCII-standard) without whitespaces. This process automatically configures an htpasswd identity provider.
-<3> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.17.0`.
-<4> Optional: Specify 'optional' to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify 'required' to configure all EC2 instances to use IMDSv2 only.
+<2> When creating your cluster, you can create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an htpasswd identity provider for the `cluster-admin` user.
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:265]:
```suggestion
<2> When creating your cluster, you can create a local administrator user (`cluster-admin`) for your cluster. This automatically configures an `htpasswd` identity provider for the `cluster-admin` user.
```

?

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 9]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
+<3> You can create a custom password for the `cluster-admin` user, or have the system generate a password. If you do not create a custom password, the generated password is displayed in the command line output. If you specify a custom password, the password must be at least 14 characters (ASCII-standard) without any whitespace. Once defined, the password is hashed and transported securely.
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:266]:
```suggestion
<3> You can create a custom password for the `cluster-admin` user, or have the system generate a password. If you do not create a custom password, the generated password is displayed in the command line output. If you specify a custom password, the password must be at least 14 characters (ASCII-standard) without any whitespace. When defined, the password is hashed and transported securely.
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 10]
```diff
@@ -222,48 +222,50 @@ Any optional fields can be left empty and a default will be selected.
...
+<4> When creating the cluster, the listed `OpenShift version` options include the major, minor, and patch versions, for example `4.17.0`.
+<5> Optional: Specify 'optional' to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify 'required' to configure all EC2 instances to use IMDSv2 only.
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:268]:
```suggestion
<5> Optional: Specify `optional` to configure all EC2 instances to use both v1 and v2 endpoints of EC2 Instance Metadata Service (IMDS). This is the default value. Specify `required` to configure all EC2 instances to use IMDSv2 only.
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 11]
```diff
@@ -272,15 +274,15 @@ The Instance Metadata Service settings cannot be changed after your cluster is c
 ====
 endif::openshift-rosa[]
 +
-<5> If you have more than one set of account roles for your cluster version in your AWS account, an interactive list of options is provided.
-<6> Optional: Specify an unique identifier that is passed by {product-title} and the OpenShift installer when an account role is assumed. This option is only required for custom account roles that expect an external ID.
-<7> By default, the cluster-specific Operator role names are prefixed with the cluster name and a random 4-digit hash. You can optionally specify a custom prefix to replace `<cluster_name>-<hash>` in the role names. The prefix is applied when you create the cluster-specific Operator IAM roles. For information about the prefix, see _Defining an Operator IAM role prefix_.
+<6> If you have more than one set of account roles for your cluster version in your AWS account, an interactive list of options is provided.
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:277]:
```suggestion
<6> If you have more than one set of account roles for your cluster version in your {AWS} account, an interactive list of options is provided.
```

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 12]
```diff
@@ -272,15 +274,15 @@ The Instance Metadata Service settings cannot be changed after your cluster is c
...
+<7> Optional: Specify an unique identifier that is passed by {product-title} and the OpenShift installer when an account role is assumed. This option is only required for custom account roles that expect an external ID.
+<8> By default, the cluster-specific Operator role names are prefixed with the cluster name and a random 4-digit hash. You can optionally specify a custom prefix to replace `<cluster_name>-<hash>` in the role names. The prefix is applied when you create the cluster-specific Operator IAM roles. For information about the prefix, see _Defining an Operator IAM role prefix_.
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:279]: I was expecting "Defining an Operator IAM role prefix" to be in the same document/doc book, but it is not. Maybe at a link in an add. resources section?

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc [hunk 13]
```diff
@@ -272,15 +274,15 @@ The Instance Metadata Service settings cannot be changed after your cluster is c
...
 +
 [NOTE]
 ====
 If you specified custom ARN paths when you created the associated account-wide roles, the custom path is automatically detected. The custom path is applied to the cluster-specific Operator roles when you create them in a later step.
 ====
-<8> Optional: Specify a tag that is used on all resources created by {product-title} in AWS. Tags can help you manage, identify, organize, search for, and filter resources within AWS. Tags are comma separated, for example: "key value, data input".
+<9> Optional: Specify a tag that is used on all resources created by {product-title} in AWS. Tags can help you manage, identify, organize, search for, and filter resources within AWS. Tags are comma separated, for example: "key value, data input".
```
Comment (dfitzmau) [inline modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:285]:
```suggestion
<9> Optional: Specify a tag that is used on all resources created by {product-title} in AWS. Tags can help you manage, identify, organize, search for, and filter resources within AWS. Tags are comma separated, for example: `key value, data input`.
```

**Review by dfitzmau — APPROVED** [review]
/remove-label peer-review-in-progress

/label peer-review-done

File: modules/rosa-classic-cluster-terraform-file-creation.adoc [hunk 14]
```diff
@@ -102,11 +101,25 @@ module "rosa-classic" {
...
+# Create an administrator user and define your own password
+# by uncommenting and editing the values of the following parameters:
+#  admin_credentials_username = <username>
+#  admin_credentials_password = <password>
+    
   depends_on = [time_sleep.wait_60_seconds]
 }
 EOF
 ----
+<.> Optional: Create an administrator user during cluster creation by uncommenting the appropriate parameters and editing their values if required.
```
[the remark this answers is not in the document]
Comment (laubai) [reply modules/rosa-classic-cluster-terraform-file-creation.adoc:122]: Done, thankyou!

File: modules/rosa-create-objects.adoc (hunk 4 above)
Comment (laubai) [reply modules/rosa-create-objects.adoc:208]: This doesn't match the rest of the documentation; if we need to make this change, I'd like to do it globally.

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc (hunk 5 above)
Comment (laubai) [reply modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:227]: Roger that, thankyou!

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc (hunk 7 above)
Comment (laubai) [reply modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:258]: After integrating some changes in main, it's now `<16>`

File: modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc (hunk 12 above)
Comment (laubai) [reply modules/rosa-sts-creating-a-cluster-with-customizations-cli.adoc:279]: Cheers, I've corrected the title and added to the existing additional resources section

**Review by xenolinux — COMMENTED** [review]
Merge review LGTM

**Pull request merged**